core/platform/search/elastic_search_services.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import json
+
+import feconf
+import python_utils
+
+import elasticsearch
+
+# https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-max-result-window
+# This is the maximum number of results to search through using the from and
+# size parameters to search through any given index. This number is equal to the
+# size 'index.max_result_window' and defines the maximum the sum of the
+# parameters (size + from) can be. If needed, we will have to change this to use
+# the search_after parameter or scrolling option as mentioned in the link above.
+MAXIMUM_NUMBER_OF_RESULTS = 10000
+ES = elasticsearch.Elasticsearch()
+
+
+def add_documents_to_index(documents, index_name):
+    """Adds a document to an index.
+
+    Args:
+        documents: list(dict). Each document should be a dictionary. Every key
+            in the document is a field name, and the corresponding value will be
+            the field's value. There MUST be a key named 'id', its value will be
+            used as the document's id.
+        index_name: str. The name of the index to insert the document into.
+
+    Raises:
+        Exception. A document cannot be added to the index.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+
+    for document in documents:
+        assert 'id' in document
+    for document in documents:
+        response = ES.index(
+            index_name, document, id=document['id'])
+        if response is None or response['_shards']['failed'] > 0:
+            raise Exception(
+                'Failed to add document to index.')
+
+
+def delete_documents_from_index(doc_ids, index_name):
+    """Deletes documents from an index.
+
+    Args:
+        doc_ids: list(str). A list of document ids of documents to be deleted
+            from the index.
+        index_name: str. The name of the index to delete the document from.
+
+    Raises:
+        Exception. Document id does not exist.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+    for doc_id in doc_ids:
+        assert isinstance(doc_id, python_utils.BASESTRING)
+
+    for doc_id in doc_ids:
+        if ES.exists(index_name, doc_id):
+            ES.delete(index_name, doc_id)
+        else:
+            raise Exception(
+                'Document id does not exist: %s' % doc_id)
+
+
+def clear_index(index_name):
+    """Clears an index on the elastic search instance.
+
+    Args:
+        index_name: str. The name of the index to clear.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+    # More details on clearing an index can be found here:
+    # https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch.Elasticsearch.delete_by_query
+    ES.delete_by_query(
+        index_name,
+        {
+            'query':
+                {
+                    'match_all': {}
+                }
+        })
```

Comment:
Could you use `ES.delete(index_name, '*')` here I think it would be much more simple to understand.

Reply:
I don't think that is a valid Query DSL query. This is the one that was recommended fro deleting everything from an index and prior to actually tsting the API, i think the best course of action is to use the recommendations.

Reply:
I think it shoud work (see https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-index.html) `'*'` and `'_all'` should work. If you have some link to the recommendation for `delete_by_query` I would be fine with that but I could find both options when searching for this.

Reply:
The link you mentioned above is a solution to delete all the indices in the cluster, not every document in a single index. Here is the link I used:
https://stackoverflow.com/questions/57778438/delete-all-documents-from-elasticsearch-index-in-python-3-x